src/parent/services/ParentService.js

```diff
@@ -0,0 +1,47 @@
+import configService from '../../shared/ConfigService';
+import { close } from '../actions/parent';
+
+class ParentService {
+
+    constructor(store) {
+        this.store = store;
+        this.onChildClosed = this.onChildClosed.bind(this);
+    }
+
+    onChildClosed({ name }) {
+        this.store.dispatch(close(name));
```

Comment:
Looks like we can fix the problem I mentioned in my last comment if we do this instead
```
if (Object.keys(this.store.getState()).length === 1) {
    fin.desktop.Window.getCurrent().contentWindow.close();
} else {
    this.store.dispatch(close(name));
}
```
With the persistence of state stuff being in, we don't want to dispatch the close action when the final child window is closed.